appengine/genetics/js/visualization.js

```diff
@@ -226,86 +313,210 @@ Genetics.Visualization.resetChartData_ = function() {
+      div.style.width = (div.parentNode.offsetWidth - 2) + 'px';
+      div.style.height = (div.parentNode.offsetHeight - 2) + 'px';
+  }
+
+  // Remove child DOM elements on display div.
+  goog.dom.removeChildren(Genetics.Visualization.display_);
+};
 
 /**
  * Start the visualization running.
  */
 Genetics.Visualization.start = function() {
+  Genetics.Visualization.gameOverReached_ = false;
+  Genetics.Visualization.gameRankings = {};
+  Genetics.Visualization.gameOverCause = '';
+  Genetics.Visualization.stopped_ = false;
   Genetics.Visualization.update();
 };
 
 /**
- * Start the visualization running.
+ * Update the visualization.
  */
 Genetics.Visualization.update = function() {
-  Genetics.Visualization.display_();
-  Genetics.Visualization.processCageEvents_();
-  // Frame done.  Calculate the actual elapsed time and schedule the next frame.
-  var now = Date.now();
-  var workTime = now - Genetics.Visualization.lastFrame -
-      Genetics.Visualization.lastDelay;
-  var delay = Math.max(1, (1000 / Genetics.Visualization.FPS) - workTime);
-  Genetics.Visualization.pid = setTimeout(Genetics.Visualization.update, delay);
-  Genetics.Visualization.lastFrame = now;
-  Genetics.Visualization.lastDelay = delay;
+  if (!Genetics.Visualization.stopped_) {
+    if (!Genetics.Visualization.gameOverReached_) {
+      Genetics.Visualization.processCageEvents_();
+      Genetics.Visualization.drawCharts_();
+    } else {
+      // Check that no mice are currently busy performing an animation sequence.
+      var allMiceFree = true;
+      for (var mouseID in Genetics.Visualization.mice_) {
+        var mouseAvatar = Genetics.Visualization.mice_[mouseID];
+        if (mouseAvatar.busy) {
+          allMiceFree = false;
+          break;
+        }
+      }
+      if (allMiceFree) {
+        // Display end game.
+        Genetics.Visualization.displayGameEnd();
+        return;
+      }
+    }
+    Genetics.Visualization.pid_ = setTimeout(Genetics.Visualization.update,
+        Genetics.Visualization.UPDATE_DELAY_MSC);
+  }
 };
 
 /**
  * Whether the charts need to be redrawn because they were updated.
  * @type {boolean}
  * @private
  */
-Genetics.Visualization.chartsUpdated_ = true;
+Genetics.Visualization.chartsNeedUpdate_ = true;
+
+/**
+ *
+ * @type {!Array.<!Object.<HTMLDivElement>>}
+ * @private
+ */
+Genetics.Visualization.playerStatsDivs_ = [];
+
+/***
+ * Update the percentage bars for player functions under the visualization.
+ * @private
+ */
+Genetics.Visualization.updateStats_ = function() {
+  var mouseCount = Genetics.Visualization.mouseSexes_[Genetics.Mouse.Sex.MALE] +
+      Genetics.Visualization.mouseSexes_[Genetics.Mouse.Sex.FEMALE];
+  for (var playerId = 0;
+       playerId < Genetics.Visualization.playerStatsDivs_.length; playerId++) {
+    var playerStats = Genetics.Visualization.playerStatsDivs_[playerId];
+    var pickFightPercent = 100 *
+        Genetics.Visualization.pickFightOwners_[playerId] / mouseCount || 0;
```

Comment:
Can you add one set of parens to this line (and the two others like it below).  I'm unclear on order of operations and thus what defaults to "|| 0".